addons/l10n_in/models/res_partner.py

```diff
@@ -1,17 +1,73 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+import re
+
 from odoo import api, fields, models, _
-from odoo.exceptions import ValidationError
+from odoo.addons.base_vat.models.res_partner import _ref_vat
+
 
+_ref_vat.update({'in': "12AAAAA1234AAZA"})
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
     # Use in view attrs. Need to required state_id if Country is India.
-    l10n_in_country_code = fields.Char(related="country_id.code", string="Country code")
+    l10n_in_company_country_code = fields.Char(related="property_account_payable_id.company_id.country_id.code", string="Country code")
+    l10n_in_gst_treatment = fields.Selection([
+        ('regular', 'Registered Business - Regular'),
+        ('composition', 'Registered Business - Composition'),
+        ('unregistered', 'Unregistered Business'),
+        ('consumer', 'Consumer'),
+        ('overseas', 'Overseas'),
+        ('special_economic_zone', 'Special Economic Zone'),
+        ('deemed_export', 'Deemed Export'),
+    ],string="GST Treatment")
+    l10n_in_pan_number = fields.Char(string="PAN Number")
+
+    @api.model
+    def check_vat_in(self, vat):
```

Comment:
This is for the GST IN number or for the VAT?  As it is good to check if you already have the correct VAT number in  base_vat.

Reply:
this is for vat number so now I add this to base_vat instead of l10n_in